oci/pull.bzl

```diff
@@ -412,97 +411,78 @@ alias(
 )
 """
 
-def _oci_alias_impl(rctx):
-    rctx.file("BUILD.bazel", content = _alias_target.format(
-        target_name = rctx.attr.target_name,
-        platform_map = {str(k): v for k, v in rctx.attr.platforms.items()},
-    ))
-
-oci_alias = repository_rule(
-    implementation = _oci_alias_impl,
-    attrs = {
-        "platforms": attr.label_keyed_string_dict(),
-        "target_name": attr.string(),
-    },
+_SINGLE_PLATFORM_IMAGE_ALIAS = """\
+alias(
+    name = "{target_name}",
+    actual = "@{original}",
+    visibility = ["//visibility:public"],
 )
+"""
 
-_latest_build = """\
-load("@aspect_bazel_lib//lib:jq.bzl", "jq")
-load("@bazel_skylib//rules:write_file.bzl", "write_file")
+def _oci_alias_impl(rctx):
+    if rctx.attr.platforms and rctx.attr.single_platform:
+        fail("Only one of 'platforms' or 'single_platform' may be set")
+    if not rctx.attr.platforms and not rctx.attr.single_platform:
+        fail("One of 'platforms' or 'single_platform' must be set")
+
+    if _is_tag(rctx.attr.identifier) and rctx.attr.reproducible:
+        manifest, _ = _download_manifest(rctx, rctx.attr.identifier, "mf.json")
+        result = rctx.execute(["shasum", "-a", "256", "mf.json"])
+        if result.return_code:
+            msg = "hashsum failed: \nSTDOUT:\n%s\nSTDERR:\n%s" % (result.stdout, result.stderr)
+            fail(msg)
+
+        optional_platforms = ""
+
+        if manifest["mediaType"] in _SUPPORTED_MEDIA_TYPES["index"]:
+            platforms = []
+            for submanifest in manifest["manifests"]:
+                parts = [submanifest["platform"]["os"], submanifest["platform"]["architecture"]]
+
+                # TODO: https://github.com/bazel-contrib/rules_oci/issues/122 variant, os.features, os.version etc.
+                # if "variant" in submanifest["platform"]:
+                #     parts.append(submanifest["platform"]["variant"])
+                platforms.append('"{}"'.format("/".join(parts)))
+            optional_platforms = "'add platforms {}'".format(" ".join(platforms))
 
-jq(
-    name = "platforms",
-    srcs = ["manifest_list.json"],
-    filter = "[(.manifests // [])[] | .platform | .os + \\"/\\" + .architecture]",
-    # Print without newlines because it's too hard to indent that to fit under the generated
-    # starlark code below.
-    args = ["--compact-output"],
-)
+        # buildifier: disable=print
+        fail("""
```

Comment:
I am really confused now, the PR description says we are making it a warning message, but here we `fail`.

Do you mind just making a new PR that is minimal and doesn't have things crossed out in the PR description?

Reply:
I made it fail to keep the changes to the PR minimal.